core/src/test/java/io/questdb/cairo/CreateTableTest.java

```diff
@@ -161,6 +161,197 @@ public void testCreateTableAsSelectInheritsColumnIndex() throws Exception {
+                {"x", "SYMBOL"},
+                {"z", "STRING"},
+                {"y", "BOOLEAN"}
+        };
+
+        assertCompile("create table x (" +
+                getColumnDefinitions(columnTypes) + ")"
+        );
+        assertQuery("a\tb\tc\td\te\tf\tg\th\tt\tx\tz\ty\n", "select * from tab", "create table tab (like x)", null);
+        assertColumnTypes("tab", columnTypes);
+    }
+
+    private void assertColumnTypes(String tableName, String[][] columnTypes) throws Exception {
+        assertMemoryLeak(() -> {
+            try (TableReader reader = engine.getReader(AllowAllCairoSecurityContext.INSTANCE, tableName)) {
+                TableReaderMetadata metadata = reader.getMetadata();
+                for (int i = 0; i < columnTypes.length; i++) {
+                    String[] arr = columnTypes[i];
+                    Assert.assertEquals(arr[0], metadata.getColumnName(i));
+                    Assert.assertEquals(arr[1], ColumnType.nameOf(metadata.getColumnType(i)));
+                }
+            }
+        });
+    }
+
+    private void assertWithClauseParameters(String tableName, int maxUncommittedRows, int commitLag) throws Exception {
+        assertMemoryLeak(() -> {
+            try (TableReader reader = engine.getReader(AllowAllCairoSecurityContext.INSTANCE, tableName)) {
+                Assert.assertEquals(commitLag, reader.getMetadata().getCommitLag());
+                Assert.assertEquals(maxUncommittedRows, reader.getMetadata().getMaxUncommittedRows());
+            }
+        });
+    }
+
+    private void assertWalEnabled(String tableName, boolean isWalEnabled) throws Exception {
+        assertMemoryLeak(() -> {
+            try (TableReader reader = engine.getReader(AllowAllCairoSecurityContext.INSTANCE, tableName)) {
+                Assert.assertEquals(isWalEnabled, reader.getMetadata().isWalEnabled());
+            }
+        });
+    }
+
+    private void assertSymbolParameters(String tableName, int colIndex, SymbolParameters parameters) throws Exception {
+        assertMemoryLeak(() -> {
+            try (TableReader reader = engine.getReader(AllowAllCairoSecurityContext.INSTANCE, tableName)) {
+                if (parameters.symbolCapacity != null) {
+                    Assert.assertEquals(parameters.symbolCapacity.intValue(), reader.getSymbolMapReader(colIndex).getSymbolCapacity());
+                }
+                Assert.assertEquals(parameters.isCached, reader.getSymbolMapReader(colIndex).isCached());
+                Assert.assertEquals(parameters.isIndexed, reader.getMetadata().isColumnIndexed(colIndex));
+                if (parameters.indexBlockCapacity != null) {
+                    Assert.assertEquals(parameters.indexBlockCapacity.intValue(), reader.getMetadata().getIndexValueBlockCapacity(colIndex));
+                }
+            }
+        });
+    }
+
+    private void assertPartitionIndex(String tableName, int partitionIndex, int timestampIndex) throws Exception {
```

Comment:
nit: I'd call this method `assertPartitionAndTimestamp`. Current name is slightly misleading.

Reply:
Done